joint_trajectory_controller/test/test_trajectory_actions.cpp

```diff
@@ -0,0 +1,595 @@
+    FAIL();
+  }
+
+  auto traj_lifecycle_node = traj_controller->get_lifecycle_node();
+  rclcpp::Parameter joint_parameters("joints", joint_names);
+  traj_lifecycle_node->set_parameter(joint_parameters);
+
+  rclcpp::Parameter operation_mode_parameters("write_op_modes", op_mode);
+  traj_lifecycle_node->set_parameter(operation_mode_parameters);
+
+  rclcpp::executors::MultiThreadedExecutor executor;
+  executor.add_node(traj_lifecycle_node->get_node_base_interface());
+
+  traj_controller->on_configure(traj_lifecycle_node->get_current_state());
+  traj_controller->on_activate(traj_lifecycle_node->get_current_state());
+  SetUpActionClient();
+  executor.add_node(node->get_node_base_interface());
+
+  auto future_handle = std::async(
+    std::launch::async, [&executor]() -> void {
+      executor.spin();
+    });
+
+  auto thread_func = [&]() {
+      // controller hardware cycle update loop
+      auto start_time = rclcpp::Clock().now();
+      rclcpp::Duration wait = rclcpp::Duration::from_seconds(2.0);
+      auto end_time = start_time + wait;
+      while (rclcpp::Clock().now() < end_time) {
+        test_robot->read();
+        traj_controller->update();
+        test_robot->write();
+      }
+    };
+  std::thread controller_hw_thread(thread_func);
+
+  // common_goal_response and common_result_response
+  // sometimes doesnt receive calls when we dont sleep
+  std::this_thread::sleep_for(std::chrono::milliseconds(200));
+
+  GoalOptions opt;
+  opt.goal_response_callback =
+    std::bind(&TestTrajectoryActions::common_goal_response, this, _1);
+  opt.result_callback =
+    std::bind(&TestTrajectoryActions::common_result_response, this, _1);
+  opt.feedback_callback = nullptr;
+
+  // send goal
+  {
+    std::vector<JointTrajectoryPoint> points;
+    JointTrajectoryPoint point;
+    point.time_from_start.set__sec(0);  // start asap
+    point.positions.resize(joint_names.size());
+
+    point.positions[0] = 1.0;
+    point.positions[1] = 2.0;
+    point.positions[2] = 3.0;
+
+    points.push_back(point);
+
+    sendActionGoal(points, 1, opt);
+  }
+  controller_hw_thread.join();
+
+  EXPECT_EQ(true, common_goal_accepted);
+  EXPECT_EQ(rclcpp_action::ResultCode::SUCCEEDED, common_resultcode);
+
+  EXPECT_EQ(1.0, test_robot->pos1);
+  EXPECT_EQ(2.0, test_robot->pos2);
+  EXPECT_EQ(3.0, test_robot->pos3);
+
+  // start again
+  common_goal_accepted = false;
+  common_resultcode = rclcpp_action::ResultCode::UNKNOWN;
+  controller_hw_thread = std::thread(thread_func);
+
+  // add feedback
+  bool feedback_recv = false;
+  opt.feedback_callback = [&](
+    rclcpp_action::ClientGoalHandle<FollowJointTrajectoryMsg>::SharedPtr,
+    const std::shared_ptr<const FollowJointTrajectoryMsg::Feedback> feedback)
+    {
+      (void)feedback;
+      feedback_recv = true;
+    };
+
+  // send goal with multiple points
+  {
+    std::vector<JointTrajectoryPoint> points;
+    JointTrajectoryPoint point1;
+    point1.time_from_start.set__sec(0);
```

Comment:
Is there a one-line with rclcpp::Time initializing it from chrono to set these fields instead?

Reply:
replaced with rclcpp::Duration::from_seconds(0.0) 2b55538